src/Microsoft.AspNetCore.Razor.Language/TagHelperBinding.cs

```diff
@@ -3,20 +3,36 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.AspNetCore.Razor.Language.Legacy
+namespace Microsoft.AspNetCore.Razor.Language
 {
     public sealed class TagHelperBinding
     {
         public IReadOnlyDictionary<TagHelperDescriptor, IEnumerable<TagMatchingRule>> _mappings;
 
-        internal TagHelperBinding(IReadOnlyDictionary<TagHelperDescriptor, IEnumerable<TagMatchingRule>> mappings)
+        internal TagHelperBinding(
+            string tagName,
+            IEnumerable<KeyValuePair<string, string>> attributes,
+            string parentTagName,
+            IReadOnlyDictionary<TagHelperDescriptor, IEnumerable<TagMatchingRule>> mappings)
         {
+            TagName = TagName;
+            Attributes = attributes;
+            ParentTagName = parentTagName;
+
             _mappings = mappings;
             Descriptors = _mappings.Keys;
         }
 
         public IEnumerable<TagHelperDescriptor> Descriptors { get; }
 
+        public string TagName { get; }
+
+        public string ParentTagName { get; }
+
+        public IEnumerable<KeyValuePair<string, string>> Attributes { get; }
```

Comment:
I meant exposing it as `IReadOnlyList`. But spoke offline and there is no reason to do it 👍